### PR TITLE
173248280 add amin panel

### PR DESCRIPTION
**Pull request by knowuh**
This is a work in progress, it contains bugs, and shouldn't be merged until we resolve an issue with project references.

@scytacki

### CODE REVIEW

**Comment by scytacki** [comment]
@knowuh did you consider running the react-admin-interface on the host system and communicating with the docker version of the graphql-backend?
That seems like it would make development a bit easier (if setup can be simplified) since the react-admin-interface will match the dev environment that is used in most of our other projects.

Probably dotenv can be used to load shared variables into node so things like SITE_URL can be used in the react-admin-interface configuration.

I think when we try to deploy this I would try building the react-admin-interface and putting the build products in the rails/public folder. Much like the react-components is currently working. That way we won't need a separate docker container just for this static site.

**Comment by knowuh** [comment]
> did you consider running the react-admin-interface on the host system and communicating with the docker version of the graphql-backend?

That is how I had it working before. It seemed easiest to share the top-level `.env` and docker-compose vars by running the development server for react-admin in a container.  So far, as a dev, I have really be enjoying this setup. `docker-compose-up` starts node-mon running and serving react-admin at `admin.portal.docker`. Nothing is preventing a dev from running `npm install` and `npm start` from the react-admin directory. I could put those instructions back in, and make these services exist in an optional docker overlay, but I am not sure its worth it.

> Probably dotenv can be used to load shared variables into node so things like SITE_URL can be used in the react-admin-interface configuration.

To get `.env` variables to the client, *react-scripts* injects any env variables that start with `REACT_APP_*` into the built code, available under `process.env.*`.  So I think this works pretty well for devs.

> I think when we try to deploy this I would try building the react-admin-interface and putting the build products in the rails/public folder. Much like the react-components is currently working. That way we won't need a separate docker container just for this static site.

I agree, probably for deployment we will want to bundle a built artifact. I will look at the recent portal-components work to see what the existing pattern is, and try to create something similar.

**Comment by scytacki** [comment]
Yeah I don't think it is worth prompting both ways of developing. My concern with running it in the container is that is a developer would need to remember to go into docker to update npm dependencies. Or if they are building an asset to copy into rails, this wouldn't really be possible to do inside of the docker container, so probably the easiest way to build that asset is doing an npm install on the local host machine.

However, I'm fine with leaving this as you've got and we can see how it goes. I'm pretty sure it would be easy to flip flop if we need to.

**Comment by knowuh** [comment]
@scytacki only 191 files to review …

File: rails/app/controllers/api/v1/jwt_controller.rb [hunk 1]
```diff
@@ -35,6 +35,18 @@ def portal
...
+    else
+      claims[:admin] = -1
+    end
+    claims[:project_admins] = []
+    user.project_users.each do |p|
+      if(p.is_admin)
+        claims[:project_admins].push(p.project_id)
+      end
+    end
+
```
Comment (knowuh) [inline rails/app/controllers/api/v1/jwt_controller.rb:49]: @scytacki this is probably the most dangerous change to the portal in this PR.

Comment (scytacki) [reply rails/app/controllers/api/v1/jwt_controller.rb]: Good point. I think it is safe enough to try out.

**Pull request merged**